flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableWritableCatalog.java

```diff
@@ -131,4 +137,62 @@ void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfE
 	void alterTable(ObjectPath tableName, CatalogBaseTable newTable, boolean ignoreIfNotExists)
 		throws TableNotExistException, CatalogException;
 
+	// ------ partitions ------
+
+	/**
+	 * Create a partition.
+	 *
+	 * @param tablePath path of the table.
+	 * @param partitionSpec partition spec of the partition
+	 * @param partition the partition to add.
+	 * @param ignoreIfExists flag to specify behavior if a table with the given name already exists:
+	 *                       if set to false, it throws a TableAlreadyExistException,
+	 *                       if set to true, nothing happens.
+	 *
+	 * @throws TableNotExistException thrown if the target table does not exist
+	 * @throws TableNotPartitionedException thrown if the target table is not partitioned
+	 * @throws PartitionSpecInvalidException thrown if the given partition spec is invalid
+	 * @throws PartitionAlreadyExistsException thrown if the target partition already exists
+	 * @throws CatalogException in case of any runtime exception
+	 */
+	void createPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec, CatalogPartition partition, boolean ignoreIfExists)
```

Comment:
Can you explain more about this interface? In which case this API would be invoked. I mean, does there exists some partition-only operations? My feeling is most operations to partition is through table.

Reply:
An example from a SQL perspective is "ALTER TABLE ADD PARTITION". Operations on partitions will be using and only using table as an identifier to locate the partition, and that's why this API includes tablePath.